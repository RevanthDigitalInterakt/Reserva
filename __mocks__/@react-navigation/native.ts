const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
