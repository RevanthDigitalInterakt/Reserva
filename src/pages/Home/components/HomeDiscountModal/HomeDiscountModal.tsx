import React, { useEffect, useMemo, useState } from 'react';
import DiscountCodeModal from '../../../../modules/Home/component/DiscountCodeModal';
import { useHomeStore } from '../../../../zustand/useHomeStore';

function HomeDiscountModal() {
  const { discountBar } = useHomeStore(['discountBar']);
  const [isVisible, setVisible] = useState(false);

  const showBar = useMemo(() => (
    discountBar?.titleBar
  ), [discountBar?.titleBar]);

  useEffect(() => {
    setVisible(!!showBar);
  }, [showBar]);

  if (!showBar) {
    return null;
  }

  return (
    <DiscountCodeModal
      data={{
        coupon: discountBar?.coupon || '',
        colorButton: discountBar?.colorButton || '',
        titleButton: discountBar?.titleButton || '',
        descriptionModal: discountBar?.descriptionModal || '',
        shareMessage: discountBar?.shareMessage || '',
        titleModal: discountBar?.titleModal || '',
        colorBar: discountBar?.colorBar || '',
        titleBar: discountBar?.titleBar || '',
      }}
      isVisible={isVisible}
      onClose={() => setVisible(false)}
    />
  );
}

export default HomeDiscountModal;
