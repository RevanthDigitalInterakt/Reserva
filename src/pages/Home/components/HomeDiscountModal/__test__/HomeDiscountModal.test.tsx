import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react-native';
import { theme } from '@usereservaapp/reserva-ui';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components/native';
import HomeDiscountModal from '../HomeDiscountModal';
import { HomeConfigDocument } from '../../../../../base/graphql/generated';

// MOCKS
const mockNavigate = jest.fn();

const mockHomeConfigQuery = {
  request: { query: HomeConfigDocument, variables: {} },
  result: {
    data: {
      homeConfig: {
        id: 'MainConfig',
        offersPage: 'collection:2772',
        discountCodeBar: {
          titleBar: 'Ganhe R$50 na sua Primeira Compra', colorBar: '#38A238', titleModal: '*Válido nas compras acima de R$199. Desconto aplicado automaticamente no carrinho. Não cumulativo com outras promoções. Não cumulativo nas camisetas com desconto.', descriptionModal: 'Use o cupom RSVAPP50', titleButton: 'Aproveite!', colorButton: '#38A238', shareMessage: null, coupon: null, __typename: 'ConfigDiscountBarOutput',
        },
        __typename: 'ConfigOutput',
      },
    },
  },
};

jest.mock('react-native-share', () => ({
  open: jest.fn(() => Promise.resolve()),
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockNavigate }),
}));

jest.mock('../../../../../zustand/useHomeStore', () => ({
  useHomeStore: () => ({
    onLoad: () => {},
    discountBar: mockHomeConfigQuery.result.data.homeConfig.discountCodeBar
  }),
}));

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <MockedProvider
      mocks={[mockHomeConfigQuery]}
      addTypename={false}
    >
      <HomeDiscountModal />
    </MockedProvider>
  </ThemeProvider>
);

describe('HomeDiscountModal', () => {
  beforeEach(async () => {
    jest.useFakeTimers({ legacyFakeTimers: true });
    render(TestingComponent);
  });

  it('should render properly', () => {
    const card = screen.getByTestId('com.usereserva:id/discount_code_modal_container');
    expect(card).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
