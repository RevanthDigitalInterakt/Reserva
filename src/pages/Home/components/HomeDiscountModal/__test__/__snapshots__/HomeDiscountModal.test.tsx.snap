// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDiscountModal should match to snapshot 1`] = `
[
  <View
    backgroundColor="#38A238"
    flexDirection="row"
    height={37}
    style={
      {
        "backgroundColor": "#38A238",
        "flexDirection": "row",
        "height": 37,
        "width": "100%",
      }
    }
    testID="com.usereserva:id/discount_code_modal_container"
    width="100%"
  >
    <View
      alignItems="center"
      flex={1}
      justifyContent="center"
      style={
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        alignItems="center"
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            color="white"
            fontFamily="reservaSansRegular"
            fontSize={13}
            style={
              {
                "color": "#FFF",
                "fontFamily": "ReservaSans-Regular",
                "fontSize": 13,
              }
            }
            testID="com.usereserva:id/discout_code_modal_title"
          >
            Ganhe R$50 na sua Primeira Compra
          </Text>
        </View>
      </View>
    </View>
    <View
      alignItems="center"
      justifyContent="center"
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 28,
        }
      }
      width={28}
    >
      <View
        alignItems="center"
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 30,
              "left": 30,
              "right": 30,
              "top": 30,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="com.usereserva:id/discout_code_modal_button"
        >
          <View
            flexGrow={0}
            justifyContent="flex-end"
            style={
              {
                "flexGrow": 0,
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              flexGrow={0}
              justifyContent="center"
              position="absolute"
              right={-5}
              style={
                {
                  "flexGrow": 0,
                  "justifyContent": "center",
                  "position": "absolute",
                  "right": -5,
                  "top": -5,
                  "zIndex": 1,
                }
              }
              top={-5}
              zIndex={1}
            />
            <RNSVGSvgView
              bbHeight="8"
              bbWidth="8"
              color="white"
              focusable={false}
              size={8}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#FFF",
                    "height": 8,
                    "width": 8,
                  },
                  {
                    "flex": 0,
                    "height": 8,
                    "width": 8,
                  },
                ]
              }
              tintColor="white"
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  height="100%"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "height": "100%",
                        "width": "100%",
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={8.24}
                  vbWidth={8.242}
                  width="100%"
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M5.098 4.12l2.944-2.944A.69.69 0 007.066.2L4.122 3.144 1.178.2a.69.69 0 10-.976.976L3.146 4.12.202 7.064a.69.69 0 00.976.976l2.944-2.944L7.066 8.04a.69.69 0 00.976-.976z"
                      fill={
                        {
                          "payload": 4294967295,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    testID="com.usereserva:id/discount_code_modal_second_container"
    transparent={true}
    visible={false}
  />,
]
`;
